controllers/designatecentral_controller.go

```diff
@@ -488,43 +489,46 @@ func (r *DesignateCentralReconciler) reconcileNormal(ctx context.Context, instan
 			condition.DeploymentReadyRunningMessage))
 		return ctrlResult, nil
 	}
-	instance.Status.ReadyCount = depl.GetDeployment().Status.ReadyReplicas
-
-	// verify if network attachment matches expectations
-	networkReady := false
-	networkAttachmentStatus := map[string][]string{}
-	if *(instance.Spec.Replicas) > 0 {
-		networkReady, networkAttachmentStatus, err = nad.VerifyNetworkStatusFromAnnotation(
-			ctx,
-			helper,
-			instance.Spec.NetworkAttachments,
-			serviceLabels,
-			instance.Status.ReadyCount,
-		)
-		if err != nil {
-			return ctrl.Result{}, err
+
+	if depl.GetDeployment().Generation <= depl.GetDeployment().Status.ObservedGeneration {
```

Comment:
I would not declare success on the < case

Reply:
We're not returning success, we're just proceeding in the reconciliation loop. I'm not sure we need to explicitly handle the `<` case and return.
In theory we can continue the reconciliation loop if `<=`, and not stop. The deployement controller is supposed to handle any inconsistencies before it returns Ready to our controller.

Reply:
Otherwise let's go back to `==` condition and let the current reconciliation return false w/o the need of necessarily handling the `<` case

Reply:
Sorry I wasn't precise enough. By allowing to consider the ReadyReplicas in the case of < the controller might declare success on a status that has a newer generation than what this controller requested. If that spec generation introduced a change that this controller does not agree with then the controller should not accept the status of the reconciliation of the spec version.

> The deployement controller is supposed to handle any inconsistencies before it returns Ready to our controller.

the deployment controller will return ready for a spec generation that is not set by this controller so we should ignore that readyness until we can make sure that such spec generation does not contradict with the spec this controller wants.

> Otherwise let's go back to == condition and let the current reconciliation return false w/o the need of necessarily handling the < case

That sounds good to me.

Reply:
Done, closing the thread as we switched to `==`